app/components/audio-player/audio-player.tsx

```diff
@@ -104,6 +107,34 @@ export function AudioPlayer({
     loadAndPlay(fileUri, track.webUri, sound, { shouldPlay: true }, onPlaybackStatusUpdate)
   }, [fileUri, track.webUri, sound, onPlaybackStatusUpdate])
 
+  const currentDownloadState = useRef(downloadState)
+
+  useEffect(() => {
+    if (downloadState.status !== currentDownloadState.current.status) {
+      __DEV__ &&
+        console.log(
+          `we switched from ${currentDownloadState.current.status} to ${downloadState.status}`,
+        )
+    }
+
+    // TODO: Move this to the response of `useDownload`?
+    // TODO: Make use of subscribe and a listener which sends old + new state?
+    // This would avoid the keeping the state in a ref.
+    if (
+      currentDownloadState.current.status === "DOWNLOADING" &&
+      downloadState.status === "DOWNLOADED"
+    ) {
+      switchSource(downloadState.uri)
+    } else if (
+      currentDownloadState.current.status === "DOWNLOADED" &&
+      downloadState.status === "NOT_DOWNLOADED"
+    ) {
+      switchSource(track.webUri)
+    }
+
+    currentDownloadState.current = downloadState
+  }, [downloadState.status])
```

Comment:
`switchSource` needs to be listed here because it is used in the effect function. And `switchSource` probably needs to be wrapped in `useCallback` then. See https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies and the note in https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect and https://reactjs.org/docs/hooks-reference.html#usecallback and https://github.com/facebook/react/issues/14920